### DISCUSSION TITLE
Support for yarl.URL and/or pydantic.HttpUrl

### DISCUSSION THREAD

**Discussion by joyanedel** [fastapi/typer#989] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
import pandas as pd
from yarl import URL

def main(csv_url: URL):
    chunked_dataframe = pd.read_csv(str(csv_remote_path), ...)
    ...

if __name__ == '__main__':
    typer.run(main)
```

### Description

* Add support for yarl URL and/or pydantic HttpUrl as python-decouple package parser currently does.
To me, if a variable as url as suffix, it means that it must be treated as an URL, therefore, typing it with either yarl or pydantic would be an awesome feature to have

Currently, Typer would raise the following exception for both cases

### yarl
```python
RuntimeError: Type not yet supported: <class 'yarl.URL'>
```

### pydantinc
```python
RuntimeError: Type not yet supported: <class 'pydantic_core._pydantic_core.Url'>
```

### Operating System

macOS

### Typer Version

0.12.5

### Python Version

3.12

**Comment by YuriiMotov** [comment, marked as answer]
You can make it working by specifying a parser for this type: https://typer.tiangolo.com/tutorial/parameter-types/custom-types/#type-parser

**Reply by joyanedel** [reply]
Thanks. Didn't even realize they added custom types in parameters
